### ISSUE TITLE
Sybil Attacker Report

### ISSUE THREAD

**Issue by GreyPK** [hop-protocol/hop-airdrop#177]
# Related Addresses

20 related addresses

```
0x116762385f981111591c6880e1c1db74b819812d
0x002c77a71d54e6da4b28690fe1676752a74a67eb
0x037b9e14191238dbfc4e670497b1d4ebf51373ea
0xa5d67c709a93396a91a038dbc0e17a81115a2b1a
0x7e0207d0c54cb91763d2d3bcc9a6cce349883409
0x73fe6db3b797ea23e0e5433541d6e179a896c697
0xdf70720c5a18b49748375670cf87c6d12e78b8bf
0x1c79e97c1ecbf8268f4b60564f37ddd174d8aec7
0xbd3fdcb5db40a00cccaf13a1466ebd5e73103574
0x2e05bbb43136189d98b9d3f18f1fcaaaf0a8c1ca
0x083397bf566d24d6aa2704531ef33e7c96df9068
0x74bf239a4f8abe85d3f9382b22e008ae50229ff5
0xafac76d3dd6b3744fea17975792ccb7993335352
0x5e1d4b9eba99fcbe8cce969f52ad5d8a2db31df5
0x471deb36b700b6ebab3da6573eedc2db8ed1af89
0x6f35109ddc4f3d1111b2ed82273b07755a1e923a
0x5fe933af37085ccc536a97799d6d6e83c11db76f
0x73f46d7aec489b47f7fff2233c53e82184d39e7f
0x3ae06234f9b31f37d3ec062d24ca40b7a8db9024
0xbf3dc018ddf1928d40a58e18345c8a1021acbbd6
```

# Reasoning

1) These addresses often make transactions between themselves.
2) All 20 wallets first time used HOP with swapAndSend in same day (25-27 Nov 2021) with USDT or USDC $160-$250. 
3) All 20 wallets used HOP with swapAndSend in almost same date (20-28 Feb 2022) with WETH. Each address used HOP swapAndSend 11-19 times in this day. Each transaction value was 0.08-0.17 WETH.

I generated a graph of matic transfers on Polygon. 20 of 50 addresses used that pattern.
![177issue](https://user-images.githubusercontent.com/9148071/168456581-8c868b8f-3b4e-4efa-a891-f04ed969ebe1.png)

0x116762385f981111591c6880e1c1db74b819812d first swapAndSend 220 USDC Nov-26-2021 https://polygonscan.com/tx/0xdbc995befb6bccc81834dbdf37c3fbd08ca52196a6cf1cbaf1c2986b20ff10be
After swapAndSend 12 times 0.1 WETH Feb-20-2022 https://polygonscan.com/txs?a=0x116762385f981111591c6880e1c1db74b819812d&f=2&ps=100&p=1

0x002c77a71d54e6da4b28690fe1676752a74a67eb first swapAndSend 204 USDC Nov-26-2021
https://polygonscan.com/tx/0x520f7717e1292611827352d155e67733925e48c1f999fc264ef1c4f5feab2792
After swapAndSend 16 times 0.1 WETH Feb-21-2022 https://polygonscan.com/txs?a=0x002c77a71d54e6da4b28690fe1676752a74a67eb&f=2&ps=100&p=1

0x037b9e14191238dbfc4e670497b1d4ebf51373ea first swapAndSend 230 USDC Nov-26-2021 
https://polygonscan.com/tx/0xb0a669c4247e7739b581977a79a327ff52379feec8ed40cc56d5db2b05d0b80a
After swapAndSend 12 times 0.13 WETH Feb-28-2022 https://polygonscan.com/txs?a=0x037b9e14191238dbfc4e670497b1d4ebf51373ea&f=2&ps=100&p=1

0xa5d67c709a93396a91a038dbc0e17a81115a2b1a first swapAndSend 174 USDC Nov-26-2021 https://polygonscan.com/tx/0x12dd0c455820617a716373283b6500413b078f6668c64fbe3531dbf3a39596f1
After swapAndSend 12 times 0.09 WETH Feb-21-2022 https://polygonscan.com/txs?a=0xa5d67c709a93396a91a038dbc0e17a81115a2b1a&f=2&ps=100&p=1

0x7e0207d0c54cb91763d2d3bcc9a6cce349883409 first swapAndSend 234 USDC Nov-26-2021
https://polygonscan.com/tx/0xa9935755b0b68b0bca6ce7b738528b4715520f8161a876a58ccae2c8950b335f
After swapAndSend 11 times 0.1 WETH Feb-20-2022 https://polygonscan.com/txs?a=0x7e0207d0c54cb91763d2d3bcc9a6cce349883409&f=2&ps=100&p=1

0x73fe6db3b797ea23e0e5433541d6e179a896c697 first swapAndSend 239 USDC Nov-26-2021 https://polygonscan.com/tx/0xe631ea01dc9096932744d5edb88f05fae458702503648c9b86a4ada81c41f8da
After swapAndSend 21 times 0.16 WETH Feb-25-2022 https://polygonscan.com/txs?a=0x73fe6db3b797ea23e0e5433541d6e179a896c697&f=2&ps=100&p=1

0xdf70720c5a18b49748375670cf87c6d12e78b8bf first swapAndSend 226 USDT Nov-26-2021
https://polygonscan.com/tx/0x35427a9f33f17c6017c271f385f0f4f2725267d2c90237d56f897c0c78a1f205
After swapAndSend 13 times 0.14 WETH Feb-27-2022 https://polygonscan.com/txs?a=0xdf70720c5a18b49748375670cf87c6d12e78b8bf&f=2&ps=100&p=1

0x1c79e97c1ecbf8268f4b60564f37ddd174d8aec7 first swapAndSend 242 USDC Nov-26-2021
https://polygonscan.com/tx/0x5bdbd3234ae4d86a4c69dab853a7a28a801d12cbd88d335081b3a1e7961c0ef8
After swapAndSend 17 times 0.17 WETH Feb-24-2022 https://polygonscan.com/txs?a=0x1c79e97c1ecbf8268f4b60564f37ddd174d8aec7&f=2&ps=100&p=1

0xbd3fdcb5db40a00cccaf13a1466ebd5e73103574 first swapAndSend 227 USDC Nov-26-2021 https://polygonscan.com/tx/0x91b0d65fd2eccb5c9fd47f1bea62ccaf2638bb08ad5b5078a63b11c0d2f31ca5
After swapAndSend 14 times 0.13 WETH Feb-28-2022 https://polygonscan.com/txs?a=0xbd3fdcb5db40a00cccaf13a1466ebd5e73103574&f=2&ps=100&p=1

0x2e05bbb43136189d98b9d3f18f1fcaaaf0a8c1ca first swapAndSend 163 USDC Nov-26-2021
https://polygonscan.com/tx/0x3e30f504ab59a9e020c4141f58bc468b44d86b8dad792360195c53e4705d74a6
After swapAndSend 19 times 0.08 WETH Feb-22-2022 https://polygonscan.com/txs?a=0x2e05bbb43136189d98b9d3f18f1fcaaaf0a8c1ca&f=2&ps=100&p=1

0x083397bf566d24d6aa2704531ef33e7c96df9068 swapAndSend ~$209 Nov-27-2021 and back
After swapAndSend 10 times 0.09 WETH Feb-16-2022

0x74bf239a4f8abe85d3f9382b22e008ae50229ff5 swapAndSend ~$250 Nov-25-2021 and back
After swapAndSend 18 times 0.1 WETH Feb-16-2022

0xafac76d3dd6b3744fea17975792ccb7993335352 swapAndSend ~$230 Nov-25-2021 and back
After swapAndSend 14 times 0.1 WETH Feb-15-2022

0x5e1d4b9eba99fcbe8cce969f52ad5d8a2db31df5 swapAndSend ~$220 Nov-25-2021 and back
After swapAndSend 12 times 0.1 WETH Feb-14-2022

0x471deb36b700b6ebab3da6573eedc2db8ed1af89 swapAndSend ~$130 Nov-25-2021 and back
After swapAndSend 18 times 0.12 WETH Feb-13-2022

0x6f35109ddc4f3d1111b2ed82273b07755a1e923a swapAndSend ~$200 Nov-27-2021 and back
After swapAndSend 10 times 0.1 WETH Feb-16-2022

0x5fe933af37085ccc536a97799d6d6e83c11db76f swapAndSend ~$220 Nov-27-2021 and back
After swapAndSend 12 times 0.1 WETH Feb-17-2022

0x73f46d7aec489b47f7fff2233c53e82184d39e7f swapAndSend ~$230 Nov-27-2021 and back
After swapAndSend 7 times 0.1 WETH Feb-18-2022

0x3ae06234f9b31f37d3ec062d24ca40b7a8db9024 swapAndSend ~$220 Nov-27-2021 and back
After swapAndSend 10 times 0.1 WETH Feb-19-2022

0xbf3dc018ddf1928d40a58e18345c8a1021acbbd6 swapAndSend ~$230 Nov-27-2021 and back
After swapAndSend 11 times 0.1 WETH Feb-20-2022
# Methodology

I checked polygon transactions with polygonscan API and filtered transactions, where 10 recipients addresses are eligible for the airdrop, that make transactions between themselves. Then I checked when and how these wallets interact with HOP. Node js code available on request.

# Rewards Address

0x7d3f6048091aEdE6198736b8C6ac527c314e25ff

**Comment by xbwl20** [comment]
审题了吗。20个以上的都看不懂啊？

**Comment by GreyPK** [comment]
Sorry I don't understand. In discord community mod said that it can be published anyway, the team will look into it and consider

**Comment by GreyPK** [comment]
I generated graph of matic transfers on Polygon and find 10 more related addresses. So 20 related addresses now.

**Comment by zlgitol** [comment]
I do not think these can prove that no legitimate users are included.  
No significant similar behaviors are shown here. 
You can't just bring neighbors of the reported address to enlarge the address list. 
There are always neighbors.

**Comment by GreyPK** [comment]
The specified addresses have the same behavior pattern. use of HOP bridge between Polygon and Gnosis, WETH 7+ times in this day, each transaction value was 0.08-0.17 WETH, 20-28 Feb 2022.

https://explorer.hop.exchange/?account=0x116762385f981111591c6880e1c1db74b819812d
https://explorer.hop.exchange/?account=0x002c77a71d54e6da4b28690fe1676752a74a67eb
https://explorer.hop.exchange/?account=0x037b9e14191238dbfc4e670497b1d4ebf51373ea
https://explorer.hop.exchange/?account=0xa5d67c709a93396a91a038dbc0e17a81115a2b1a
https://explorer.hop.exchange/?account=0x7e0207d0c54cb91763d2d3bcc9a6cce349883409
https://explorer.hop.exchange/?account=0x73fe6db3b797ea23e0e5433541d6e179a896c697
https://explorer.hop.exchange/?account=0xdf70720c5a18b49748375670cf87c6d12e78b8bf
https://explorer.hop.exchange/?account=0x1c79e97c1ecbf8268f4b60564f37ddd174d8aec7
https://explorer.hop.exchange/?account=0xbd3fdcb5db40a00cccaf13a1466ebd5e73103574
https://explorer.hop.exchange/?account=0x2e05bbb43136189d98b9d3f18f1fcaaaf0a8c1ca
https://explorer.hop.exchange/?account=0x083397bf566d24d6aa2704531ef33e7c96df9068
https://explorer.hop.exchange/?account=0x74bf239a4f8abe85d3f9382b22e008ae50229ff5
https://explorer.hop.exchange/?account=0xafac76d3dd6b3744fea17975792ccb7993335352
https://explorer.hop.exchange/?account=0x5e1d4b9eba99fcbe8cce969f52ad5d8a2db31df5
https://explorer.hop.exchange/?account=0x471deb36b700b6ebab3da6573eedc2db8ed1af89
https://explorer.hop.exchange/?account=0x6f35109ddc4f3d1111b2ed82273b07755a1e923a
https://explorer.hop.exchange/?account=0x5fe933af37085ccc536a97799d6d6e83c11db76f
https://explorer.hop.exchange/?account=0x73f46d7aec489b47f7fff2233c53e82184d39e7f
https://explorer.hop.exchange/?account=0x3ae06234f9b31f37d3ec062d24ca40b7a8db9024
https://explorer.hop.exchange/?account=0xbf3dc018ddf1928d40a58e18345c8a1021acbbd6

**Comment by shanefontaine** [comment]
Thank you for your report @GreyPK. We have verified that the addresses in this report are Sybil attackers.

The report included 20 eligible addresses as Sybil attackers which means you are eligible for 6628.753622270106756523 HOP! When Hop DAO is live, we will make a proposal for this reward — subject to a 1 year lockup, as mentioned in the [original Mirror post](https://hop.mirror.xyz/AI5fOUR0X_l0mktShDOx3mwr-hsB24gp8GvTWtS-MBc).

The qualified addresses are as follows:

```
0x116762385f981111591c6880e1c1db74b819812d
0x002c77a71d54e6da4b28690fe1676752a74a67eb
0x037b9e14191238dbfc4e670497b1d4ebf51373ea
0xa5d67c709a93396a91a038dbc0e17a81115a2b1a
0x7e0207d0c54cb91763d2d3bcc9a6cce349883409
0x73fe6db3b797ea23e0e5433541d6e179a896c697
0xdf70720c5a18b49748375670cf87c6d12e78b8bf
0x1c79e97c1ecbf8268f4b60564f37ddd174d8aec7
0xbd3fdcb5db40a00cccaf13a1466ebd5e73103574
0x2e05bbb43136189d98b9d3f18f1fcaaaf0a8c1ca
0x083397bf566d24d6aa2704531ef33e7c96df9068
0x74bf239a4f8abe85d3f9382b22e008ae50229ff5
0xafac76d3dd6b3744fea17975792ccb7993335352
0x5e1d4b9eba99fcbe8cce969f52ad5d8a2db31df5
0x471deb36b700b6ebab3da6573eedc2db8ed1af89
0x6f35109ddc4f3d1111b2ed82273b07755a1e923a
0x5fe933af37085ccc536a97799d6d6e83c11db76f
0x73f46d7aec489b47f7fff2233c53e82184d39e7f
0x3ae06234f9b31f37d3ec062d24ca40b7a8db9024
0xbf3dc018ddf1928d40a58e18345c8a1021acbbd6
```

**Comment by shanefontaine** [comment]
@GreyPK @zlgitol Please note that the data provided is great! Additional data has been found that gives this submission a non-negligible probability of eliminating legitimate users. Some of these data points are:

1. Every address in the list is in a chain on FTM
2. The behavior on every address on Hop protocol is similar

address | ensName | total | mainnet | arbitrum | optimism | polygon | xdai | totalVolume
-- | -- | -- | -- | -- | -- | -- | -- | --
0x73fe6db3b797ea23e0e5433541d6e179a896c697 |   | 23 | 0 | 0 | 0 | 22 | 1 | $9,391.08
0x1c79e97c1ecbf8268f4b60564f37ddd174d8aec7 |   | 20 | 0 | 0 | 0 | 18 | 2 | $8,430.05
0x2e05bbb43136189d98b9d3f18f1fcaaaf0a8c1ca |   | 23 | 0 | 0 | 0 | 20 | 3 | $7,273.18
0x74bf239a4f8abe85d3f9382b22e008ae50229ff5 |   | 20 | 0 | 0 | 0 | 13 | 7 | $6,552.14
0x471deb36b700b6ebab3da6573eedc2db8ed1af89 |   | 20 | 0 | 0 | 0 | 11 | 9 | $6,507.37
0xbd3fdcb5db40a00cccaf13a1466ebd5e73103574 |   | 16 | 0 | 0 | 0 | 15 | 1 | $5,684.32
0xdf70720c5a18b49748375670cf87c6d12e78b8bf |   | 15 | 0 | 0 | 0 | 14 | 1 | $5,261.04
0xafac76d3dd6b3744fea17975792ccb7993335352 |   | 16 | 0 | 0 | 0 | 9 | 7 | $5,234.87
0x037b9e14191238dbfc4e670497b1d4ebf51373ea |   | 14 | 0 | 0 | 0 | 13 | 1 | $4,845.71
0x5e1d4b9eba99fcbe8cce969f52ad5d8a2db31df5 |   | 14 | 0 | 0 | 0 | 6 | 8 | $4,460.53
0x002c77a71d54e6da4b28690fe1676752a74a67eb |   | 18 | 0 | 0 | 0 | 17 | 1 | $4,257.51
0x5fe933af37085ccc536a97799d6d6e83c11db76f |   | 14 | 0 | 0 | 0 | 12 | 2 | $3,890.86
0xbf3dc018ddf1928d40a58e18345c8a1021acbbd6 |   | 13 | 0 | 0 | 0 | 12 | 1 | $3,845.41
0x7e0207d0c54cb91763d2d3bcc9a6cce349883409 |   | 15 | 0 | 0 | 0 | 13 | 2 | $3,794.83
0x73f46d7aec489b47f7fff2233c53e82184d39e7f |   | 13 | 0 | 0 | 0 | 11 | 2 | $3,639.73
0x116762385f981111591c6880e1c1db74b819812d |   | 14 | 0 | 0 | 0 | 13 | 1 | $3,589.65
0xa5d67c709a93396a91a038dbc0e17a81115a2b1a |   | 16 | 0 | 0 | 0 | 13 | 3 | $3,528.78
0x083397bf566d24d6aa2704531ef33e7c96df9068 |   | 12 | 0 | 0 | 0 | 11 | 1 | $3,509.10
0x3ae06234f9b31f37d3ec062d24ca40b7a8db9024 |   | 12 | 0 | 0 | 0 | 11 | 1 | $3,182.47
0x6f35109ddc4f3d1111b2ed82273b07755a1e923a |   | 15 | 0 | 0 | 0 | 14 | 1 | $1,217.81

**Comment by forweb3** [comment]
sir ，One of my, our report is finding that the rule has to be a jump to survive. However, the successor has completed more than 10 user tracking records, and there is no record of him. I cancelled his reward. And this is what I found so am I not rewarded accordingly? This address has not interacted with hop. And he did not provide other information that he interacted. I report all his reports.  
0x7d3f6048091aEdE6198736b8C6ac527c314e25ff  This is a brand new empty address, why can't he provide the address he interacts with, because he is the real empty hunter. He circumvented it with better technical isolation. So here's my reasoning, I want 25% of his reward to go to me and then the remaining 75% to all eligible real users. Just like our rules.
If the report is illegal, then the judgment should also be cancelled! If not cancelled, then I report this fake report as a sybil attack. His reward should be allocated to me.
@shanefontaine
